### ISSUE TITLE
framework::Client should track players, connections and state by default

### ISSUE THREAD

**Issue by theangryangel** [theangryangel/insim.rs#18]

**Comment by theangryangel** [comment]
Changing my mind on this one, and here’s my rational:

Each app has fundamentally different needs. If I’m a TUI app I probably want to pre-process the player names to ANSI colour codes, if I’m a web app then stripping, etc. There’s not always a good common ground. For the TUI app you probably want to pre-process the CodepageString to a regular String and then never reprocess it unless the player changes name. By storing as a CodepageString (the lowest common denominator) you end up wasting CPU cycles.

And that’s just player names. Multiply by plate names, vehicles, etc. and it gets worse.

Basically I don’t see a way without being wasteful or without providing something thats rubbish right now.
